src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,823 @@
+    const std::array<real,nstate-(dim+2)> mean_turbulence_property = compute_mean_turbulence_property(conservative_soln1, conservative_soln2);
+
+    for (int i=0;i<dim+2;i++){
+        conv_num_split_flux[i] = 0.0;
+    }
+    for (int i=dim+2;i<nstate;i++)
+    {
+        for (int flux_dim = 0; flux_dim < dim; flux_dim++){
+            conv_num_split_flux[i][flux_dim] = mean_density*mean_velocities[flux_dim]*mean_turbulence_property[i-(dim+2)];
+        }
+    }
+
+    return conv_num_split_flux;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::compute_mean_turbulence_property(const std::array<real,nstate> &conservative_soln1,
+                                   const std::array<real,nstate> &conservative_soln2) const
+{
+    std::array<real,nstate-(dim+2)> mean_turbulence_property;
+
+    const std::array<real,nstate-(dim+2)> primitive_soln1_turbulence_model = convert_conservative_to_primitive_turbulence_model(conservative_soln1); 
+    const std::array<real,nstate-(dim+2)> primitive_soln2_turbulence_model = convert_conservative_to_primitive_turbulence_model(conservative_soln2); 
+
+    for (int i=0;i<nstate-(dim+2);i++){
+        mean_turbulence_property[i] = (primitive_soln1_turbulence_model[i]+primitive_soln2_turbulence_model[i])/2.0;
+    }
+
+    return mean_turbulence_property;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convective_eigenvalues (
+    const std::array<real,nstate> &conservative_soln,
+    const dealii::Tensor<1,dim,real> &normal) const
+{
+    std::array<real,dim+2> conservative_soln_rans = extract_rans_conservative_solution(conservative_soln);
+    const dealii::Tensor<1,dim,real> vel = this->navier_stokes_physics->template compute_velocities<real>(conservative_soln_rans);
+    std::array<real,nstate> eig;
+    real vel_dot_n = 0.0;
+    for (int d=0;d<dim;++d) { vel_dot_n += vel[d]*normal[d]; };
+    for (int i=0; i<dim+2; i++) {
+        eig[i] = 0.0;
+    }
+    for (int i=dim+2; i<nstate; i++) {
+        eig[i] = vel_dot_n;
+    }
+    return eig;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::max_convective_eigenvalue (const std::array<real,nstate> &conservative_soln) const
+{
+    const std::array<real,dim+2> conservative_soln_rans = extract_rans_conservative_solution(conservative_soln);
+
+    const dealii::Tensor<1,dim,real> vel = this->navier_stokes_physics->template compute_velocities<real>(conservative_soln_rans);
+
+    const real vel2 = this->navier_stokes_physics->template compute_velocity_squared<real>(vel);
+
+    const real max_eig = sqrt(vel2);
+
+    return max_eig;
```

Comment:
same definition as euler.cpp but without the speed of sound?

Reply:
As I checked, the RANS models have convective flux terms as flow velocities times working variables, the (maximal) eigenvalues of flux Jacobian would always be the flow velocity, and the characteristic of turbulence properties is only carried by the flow.

Reply:
Ok just add the reference in the header file

Reply:
added reference